python/instrumentation/openinference-instrumentation-llama-index/src/openinference/instrumentation/llama_index/_callback.py

```diff
@@ -314,90 +330,56 @@ def on_event_end(
+        """This is intentionally empty."""
 
-            if event_type is CBEventType.LLM:
-                # Need to update the template variables attribute by extracting their values
-                # from the template events, which are sibling events preceding the LLM event.
-                # Note that the stack should be sorted in chronological order, and we are
-                # going backwards in time on the stack.
-                while dfs_stack:
-                    _, preceding_event_data = dfs_stack[-1]
-                    if (
-                        preceding_event_data.parent_id != event_data.parent_id
-                        or preceding_event_data.event_type is not CBEventType.TEMPLATING
-                    ):
-                        break
-                    dfs_stack.pop()  # remove template event from stack
-                    for payload in preceding_event_data.payloads:
-                        # Add template attributes to the LLM span to which they belong.
-                        try:
-                            attributes.update(_template_attributes(payload))
-                        except Exception:
-                            logger.exception(
-                                f"Failed to convert payload to semantic attributes. "
-                                f"event_type={preceding_event_data.event_type}, payload={payload}",
-                            )
-
-            if event_data.end_time is None:
-                self._stragglers[event_id] = event_data
-            elif not event_data.is_dispatched:
-                _finish_tracing(event_data)
-            dfs_stack.extend(adjacency_list[event_id])
-
-
-class _Stragglers(OrderedDict[_EventId, _EventData]):
+
+_Value = TypeVar("_Value")
+
+
+class _BoundedDict(OrderedDict[str, _Value]):
     """
-    Stragglers are events that have not ended before their traces have ended. If an event is in
-    this container, then we are still waiting for its `on_event_end` to be called after its trace
-    has ended. However, this call may never take place. One example is when the LLM raises an
-    exception in the following code location, in which case the LLM event never ends.
+    One use case for this is when the LLM raises an exception in the following code location, in
+    which case the LLM event will never be popped and will remain in the container forever.
     https://github.com/run-llama/llama_index/blob/dcef41ee67925cccf1ee7bb2dd386bcf0564ba29/llama_index/llms/base.py#L62
     Therefore, to prevent memory leak, this container is limited to a certain capacity, and when it
-    exceeds that capacity, the oldest straggler by insertion order is popped and are forced to
-    finish tracing.
+    reaches that capacity, the oldest item by insertion order will be popped.
     """  # noqa: E501
 
-    def __init__(self, capacity: int) -> None:
+    def __init__(self, capacity: int = 1000, fn: Optional[Callable[[_Value], None]] = None) -> None:
```

Comment:
`fn` is a bit vague. Can you re-name it? this is the callback when an item get's evicted right? `on_evict_fn` or something like that would help make this more readable

Reply:
done